core/src/main/java/org/elasticsearch/index/query/SimpleQueryParser.java

```diff
@@ -202,51 +203,135 @@ private Query newPossiblyAnalyzedQuery(String field, String termStr) {
+        private static final boolean DEFAULT_LENIENT = false;
+        private Boolean lenient = DEFAULT_LENIENT;
+
+        /** Specifies whether wildcards should be analyzed. */
+        private static final boolean DEFAULT_ANALYZE_WILDCARD = false;
+        private Boolean analyzeWildcard = DEFAULT_ANALYZE_WILDCARD;
 
+        /**
+         * Generates default {@link Settings} object (uses ROOT locale, does
+         * lowercase terms, no lenient parsing, no wildcard analysis).
+         * */
         public Settings() {
+            //
+        }
 
+        public Settings(Locale locale, Boolean lowercaseExpandedTerms, Boolean lenient, Boolean analyzeWildcard) {
+            this.locale = locale;
+            this.lowercaseExpandedTerms = lowercaseExpandedTerms;
+            this.lenient = lenient;
+            this.analyzeWildcard = analyzeWildcard;
         }
 
-        public void locale(Locale locale) {
+        /** Specifies the locale to use for parsing, Locale.ROOT by default. */
+        public Settings locale(Locale locale) {
             this.locale = locale;
+            return this;
         }
 
+        /** Returns the locale to use for parsing. */
         public Locale locale() {
             return this.locale;
         }
 
-        public void lowercaseExpandedTerms(boolean lowercaseExpandedTerms) {
+        /**
+         * Specifies whether to lowercase parse terms, defaults to true if
+         * unset.
+         */
+        public Settings lowercaseExpandedTerms(Boolean lowercaseExpandedTerms) {
             this.lowercaseExpandedTerms = lowercaseExpandedTerms;
+            return this;
         }
 
-        public boolean lowercaseExpandedTerms() {
+        /** Returns whether to lowercase parse terms. */
+        public Boolean lowercaseExpandedTerms() {
             return this.lowercaseExpandedTerms;
         }
 
-        public void lenient(boolean lenient) {
+        /** Specifies whether to use lenient parsing, defaults to false. */
+        public Settings lenient(Boolean lenient) {
             this.lenient = lenient;
+            return this;
         }
 
-        public boolean lenient() {
+        /** Returns whether to use lenient parsing. */
+        public Boolean lenient() {
             return this.lenient;
         }
 
-        public void analyzeWildcard(boolean analyzeWildcard) {
+        /** Specifies whether to analyze wildcards. Defaults to false if unset. */
+        public Settings analyzeWildcard(Boolean analyzeWildcard) {
             this.analyzeWildcard = analyzeWildcard;
+            return this;
         }
 
-        public boolean analyzeWildcard() {
+        /** Returns whether to analyze wildcards. */
+        public Boolean analyzeWildcard() {
             return analyzeWildcard;
         }
+
+        /**
+         * Sets any members that are still null to their respective default
+         * value.
+         * */
+        public void addDefaults() {
```

Comment:
why do we need this method? aren't these fields already initialized with default values, so if they changed we have the actual values otherwise the default ones?